test/python/circuit/test_circuit_registers.py

```diff
@@ -61,6 +62,16 @@ def test_qarg_string_size(self):
         """
         self.assertRaises(CircuitError, QuantumRegister, 'string')
 
+    def test_qarg_noninteger_float(self):
+        """ Test warning when passing non-integer float to QuantumRegister.
+        """
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
In many other tests we use [`assertWarns`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarns). Is there any advantage in using this style over that one?